Add auditlog documentation (#2458)
spiffe/spire: doc/auditlog.md

# Audit log

SPIRE Server can be configured to emit audit logs through the [audit_log_enabled](spire_server.md#server-configuration-file) configuration. Audit logs are sent to the same output as regular logs.

## Fields

> Each entry contains fields related with the provided `request` to each endpoint. It also contains `audit log` specific fields that provide additional information.

Reviewer: ```suggestion
Each entry contains fields related with the provided request to each endpoint. It also contains audit log specific fields that provide additional information.
```

| Key | Description | Values |
| --- | ----------- | ------ |
| type  | constant that is used to identify that current entry is an audit log | audit |
| request_id | uuid that identifies current call, it is useful for `Batch` endpoints that can emit multiple audit logs, one per each operation that is done. | |
| status | Describe if call was successfull or not | [error, success] | 
| status_code | in case of error contains `gRPC` status code  | |

> | status_message | in case of error contains error returned to caller | |

Reviewer: ```suggestion
| status_message | In case of an error, contains the error returned to the caller. | |
```

> Aditionally provides fields to to identify `caller`

Reviewer: ```suggestion
The following fields are provided to identify the caller.
```

> UDS:

Reviewer: ```suggestion
Endpoints listening on UDS:
```

> Endpoints listening on UDS:

Reviewer: ```suggestion
### Endpoints listening on UDS
```

> > **_NOTE:_**  In order to enable audit logs in K8S for UDS calls `hostPID: true` is required on SPIRE Server node.

Reviewer: ```suggestion
> **_NOTE:_**  In order to enable audit log in Kubernetes for calls done to UDS endpoints,`hostPID: true` is required in the SPIRE Server node.
```

| Key | Description                |
| ----------- | ------------------ |

> | caller_uid  | caller user ID     |

Reviewer: ```suggestion
| caller_uid  | Caller user ID.     |
```

> | caller_gid  | caller group ID    |

Reviewer: ```suggestion
| caller_gid  | Caller group ID.    |
```

> | caller_path | caller binary path |

Reviewer: ```suggestion
| caller_path | Caller binary file path. |
```

> TLS:

Reviewer: ```suggestion
Endpoints listening on TLS ports:
```

> Endpoints listening on TLS ports:

Reviewer: ```suggestion
### Endpoints listening on TLS ports
```

| Key | Description                                            | 
| --- | ------------------------------------------------------ | 

> | caller_addr | caller IP address                              |

Reviewer: ```suggestion
| caller_addr | Caller IP address.                             |
```

> | caller_id   | SPIFFE ID from certificate presented by caller |

Reviewer: ```suggestion
| caller_id   | SPIFFE ID extracted from the X.509 certificate presented by the caller. |
```
